static/js/shared/ga_events.test.tsx

```diff
@@ -682,3 +722,400 @@ describe("test ga event tool chart plot", () => {
     });
   });
 });
+describe("test ga event tool stat var click", () => {
+  test("call gtag when a stat var is selected in the stat var hierarchy", async () => {
+    // Mock gtag.
+    const mockgtag = jest.fn();
+    window.gtag = mockgtag;
+    // Mock child stat var groups.
+    axios_mock();
+
+    // When the component is rendered.
+    const statVarHierarchy = render(
+      <StatVarHierarchy {...STAT_VAR_HIERARCHY_PROPS} />
+    );
+    // Wait for the collapsible__trigger to get rendered.
+    await waitFor(() => {
+      expect(
+        statVarHierarchy.container.getElementsByClassName(
+          "Collapsible__trigger"
+        )[0]
+      ).toBeTruthy();
+    });
+
+    // Click collapsible trigger.
+    fireEvent.click(
+      statVarHierarchy.container.getElementsByClassName(
+        "Collapsible__trigger"
+      )[0]
+    );
+    const inputId = "#" + STAT_VAR_3 + "dc\\/g\\/Demographics-" + STAT_VAR_3;
+    // Wait for stat vars to get rendered.
+    await waitFor(() => {
+      expect(statVarHierarchy.container.querySelector(inputId)).toBeTruthy();
+    });
+
+    // Click the checkbox of the stat var.
+    fireEvent.click(statVarHierarchy.container.querySelector(inputId), {
+      target: { checked: true },
+    });
+    await waitFor(() => {
+      // Check gtag is called once.
+      expect(mockgtag.mock.calls.length).toEqual(1);
+      // Check the parameters passed to the gtag.
+      expect(mockgtag.mock.lastCall).toEqual([
+        "event",
+        GA_EVENT_TOOL_STAT_VAR_CLICK,
+        {
+          [GA_PARAM_STAT_VAR]: STAT_VAR_3,
+        },
+      ]);
+    });
+  });
+  test("call gtag when a place is added in the place seaerch bar", async () => {
+    const props = {
+      selectedPlace: {
+        types: null,
+      } as NamedTypedPlace,
+      enclosedPlaceType: "",
+      onPlaceSelected: () => null,
+      onEnclosedPlaceTypeSelected: () => null,
+    };
+    // Mock gtag.
+    const mockgtag = jest.fn();
+    window.gtag = mockgtag;
+    // Mock google maps.
+    window.google = {
+      maps: {
+        places: {
+          Autocomplete: jest.fn().mockImplementation((elem, opts) => {
+            return {
+              addListener: (place_changed, callback) => {
+                elem.addEventListener("change", callback);
+              },
+              getPlace: () => {
+                return { name: PLACE_ADDED };
+              },
+            };
+          }),
+        },
+      },
+    } as any;
+
+    // Render the component.
+    const placeSelector = render(<PlaceSelector {...props} />);
+    await waitFor(() =>
+      expect(placeSelector.container.querySelector("#ac")).toBeTruthy()
+    );
+
+    // Use the hardcoded result as place autocomplete.
+    fireEvent.change(placeSelector.container.querySelector("#ac"), {
+      target: { value: PLACE_ADDED },
+    });
+    await waitFor(() => expect(mockgtag.mock.calls.length).toEqual(1));
```

Comment:
add check that the gtag is called with the expected parameters?

Reply:
Thanks for catching this!